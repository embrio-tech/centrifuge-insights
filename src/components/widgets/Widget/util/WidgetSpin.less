.widget-spin {
  @apply h-full;

  .ant-spin-container {
    @apply h-full;
  }
}
