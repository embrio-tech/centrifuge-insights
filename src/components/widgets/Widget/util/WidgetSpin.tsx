import React from 'react'
import { Spin } from 'antd'

import './WidgetSpin.less'

interface WidgetSpinProps {
  spinning?: boolean
}

export const WidgetSpin: React.FC<WidgetSpinProps> = (props) => {
  const { spinning, children } = props

  return (
    <Spin spinning={spinning} wrapperClassName='widget-spin'>
      {children}
    </Spin>
  )
}
