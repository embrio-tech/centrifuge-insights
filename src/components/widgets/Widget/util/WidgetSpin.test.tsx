import React from 'react'
import renderer from 'react-test-renderer'
import { WidgetSpin } from './WidgetSpin'

describe('WidgetSpin should', () => {
  test('match snapshot!', () => {
    const tree = renderer
      .create(
        <div className='h-20'>
          <WidgetSpin spinning>Hello spinner!</WidgetSpin>
        </div>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})

export default null
