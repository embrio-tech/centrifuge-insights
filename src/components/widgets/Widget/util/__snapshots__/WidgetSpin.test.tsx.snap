// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WidgetSpin should match snapshot! 1`] = `
<div
  className="h-20"
>
  <div
    className="ant-spin-nested-loading widget-spin"
  >
    <div>
      <div
        aria-busy={true}
        aria-live="polite"
        className="ant-spin ant-spin-spinning"
      >
        <span
          className="ant-spin-dot ant-spin-dot-spin"
        >
          <i
            className="ant-spin-dot-item"
          />
          <i
            className="ant-spin-dot-item"
          />
          <i
            className="ant-spin-dot-item"
          />
          <i
            className="ant-spin-dot-item"
          />
        </span>
      </div>
    </div>
    <div
      className="ant-spin-container ant-spin-blur"
    >
      Hello spinner!
    </div>
  </div>
</div>
`;
